src/gui/tray/Window.qml

```diff
@@ -553,31 +564,41 @@ Window {
             }
         }   // Rectangle trayWindowHeaderBackground
 
-        ListView {
-            id: activityListView
-            anchors.top: trayWindowHeaderBackground.bottom
-            anchors.left: trayWindowBackground.left
-            anchors.right: trayWindowBackground.right
-            anchors.bottom: trayWindowBackground.bottom
-            clip: true
-            ScrollBar.vertical: ScrollBar {
-                id: listViewScrollbar
+       ActivityList {
+           anchors.top: trayWindowHeaderBackground.bottom
+           anchors.left: trayWindowBackground.left
+           anchors.right: trayWindowBackground.right
+           anchors.bottom: trayWindowBackground.bottom
+           
+           model: activityModel
+           onShowFileActivity: {
+               openFileActivityDialog(displayPath, absolutePath)
+           }
+           onActivityItemClicked: {
+               model.triggerDefaultAction(index)
+           }
+       }
+
+        Loader {
+            id: fileActivityDialogLoader
+
+            property string displayPath: ""
+            property string absolutePath: ""
+
+            function refresh() {
+                active = true
+                item.model.load(activityModel.accountState, absolutePath)
+                item.show()            
             }
 
-            readonly property int maxActionButtons: 2
-
-            keyNavigationEnabled: true
-
-            Accessible.role: Accessible.List
-            Accessible.name: qsTr("Activity list")
-
-            model: activityModel
+            active: false
+            sourceComponent: FileActivityDialog {
+                title: qsTr("%1 - File activity").arg(fileActivityDialogLoader.displayPath)

```

Comment:
you will also need to handle the closing of the dialog to put the loader to inactive state
something like
```
onClosing: fileActivityDialogLoader.active = false
```